api/server/server.go

```diff
@@ -112,12 +113,26 @@ func StartVolumePluginAPI(
 	return nil
 }
 
+func CheckNullClusterServerConfiguration(config ClusterServerConfiguration) {
+
+	// Set config managers to null/generic implementation if passed as null
+	if config.ConfigSecretManager == nil {
+		config.ConfigSecretManager = secrets.NewSecretManager(secrets.New())
+	}
+
+	if config.ConfigSchedManager == nil {
+		config.ConfigSchedManager = sched.NewSchedulePolicyManager(sched.New())
+	}
+
+}
+
 func StartClusterApiWithConfiguration(
```

Comment:
is this approaved as discussed with Luis , the approach

Reply:
yes, if you are talking about StartClusterApiWithConfiguration(), it's already integrated with OSD, checknul...() one is cleanup suggested by KK